src/arrayMethodJoin.js

```diff
@@ -4,9 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let resultJoin = '';
+    let counter = 0;
+
+    while (counter < length) {
+      if (counter > 0) {
+        resultJoin += `${separator}`;
+      }
+
+      let next = '';
+      const elem = this[counter];
+
+      next = elem === null || elem === undefined ? '' : elem;
```

Comment:
format ternaries properly, don't write them in one line
```suggestion
      next = elem === null || elem === undefined 
        ? '' 
        : elem;
```

Reply:
you also don't need a let here, make it a const and set value instantly
```suggestion
      const next = ...
```